style.md

```diff
@@ -1894,6 +1896,172 @@ this contract. Specifying field names inside tags makes the contract explicit,
 and it guards against accidentally breaking the contract by refactoring or
 renaming fields.
 
+### Don't fire-and-forget goroutines
+
+Goroutines are lightweight, but they're not free. They cost memory.
+Leaking goroutines slowly consume more and more memory.
```

Comment:
This makes it seem like it's the goroutine's memory usage that is the only concern, perhaps worth calling out that it's also any resources that may be referenced (E.g., timers, files, etc).

Reply:
Agree, maybe something to the effect of:

> Goroutines are lightweight, but they're not free: at a minimum, they cost memory (for their stack) and CPU (due to scheduling). While these costs are typically negligible, goroutines can cause performance issues when spawned in large numbers or without controlling their lifetimes. Additionally, because goroutines can hold pointers that prevent GC for objects visible to them, unmanaged goroutine lifetimes can lead to memory leaks.

Reply:
I mean they are *guaranteed* to consume memory, everything else is a "maybe"/likely.

Reply:
Adapted what @mway said into a something extra explaining some of the concerns.